docs/schema.yaml

```diff
@@ -0,0 +1,31 @@
+config_version: map(int(), str())
+
+general:
+  cm_device_codename: str()
+  cm_release_type: str()
+  cm_release_version: str()
+  template: str(required=False)
+
+repositories: list(include('repository'))
+
+apps: list(include('app'))
```

Comment:
It would be nice to have more validation on the apps list. I haven't seen any logical operators (at a first glance). Inside this apps list instead of just using one `include('app')` maybe we could have multiple includes or a map. And then have various app objects with different requirements that look like this:

```
app_fdroid:
  id: str()
  install_type: enum('system', 'privileged', 'user', required=False)
  repository: str(required=False)
  version: str()

app_fdroid_versioncode:
  id: str()
  install_type: enum('system', 'privileged', 'user', required=False)
  repository: str(required=False)
  versioncode: int()

app_remote:
  id: str()
  install_type: enum('system', 'privileged', 'user', required=False)
  url: str()
```